.github/workflows/codeql_security_scan.yml

```diff
@@ -0,0 +1,37 @@
+name: CodeQL Security Scan
+on:
+  push:
+  workflow_dispatch:
+
+concurrency:
+  group: codeql-s-scan-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  codeql-scan:
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v2
+        - name: Install CodeQL
+          run: |
+            cd $HOME
+            wget https://github.com/github/codeql-action/releases/download/codeql-bundle-v2.17.1/codeql-bundle-linux64.tar.gz
```

Comment:
the big question here is why we are custom installing code ql and not using the official [Github Action](https://github.com/github/codeql-action), the way that is done here has disadvantages;
1. we are hardcoding a version
2. we have to maintain it
3. we will lose the ability to see the security issues directly on Github
4. This will create a burden on the developer to create the report and share it with DM if needed

We already have a prototype for Enroll that uses this action on https://github.com/ideacrew/enroll/compare/trunk...test_codeql, and you can browse to the security tab using this branch. 

Since the GHA implementation is cleaner and less prone to errors, I don't think we should merge this as it is.

Reply:
1. it doesn't go to "issues" it goes to the security tab, and it's very easy to track who disabled and why (because it asks mandatory for a reason, I just tested it, and you can search with: `is:closed branch:test_codeql `
2. on the web it shows the item, with the snippet of code, and adds the recommendation and example. 
3. A developer not added to the organization cannot see the security items this is a small gap because we have to add DM to the repo. But it's way easier than having to generate the report locally and then share it with the DM

Reply:
After working with the CodeQL stuff you've set up I hard agree - I'm going to pull out the custom install and running my own DB/Analysis build and just use the github action helpers to do it.  I'll let you know once I merge in the fix.